app/assets/javascripts/autoform_form.js

```diff
@@ -21,24 +21,57 @@ $(document).ready(function() {
         }
         else {
             const formData = $(this).serialize();
-            $.post($(this).attr('action'), formData, function(response) {
 
-            }).complete(function(jqxhr) {
-                if (jqxhr.status === 200) {
+            $.post($(this).attr('action'), formData)
+                .done(function() {
                     $('#autoform-error').hide();
                     $('#autoform-form-error').hide();
                     $('#autoform-email').parent('.form-group').removeClass('has-error');
                     $('#autoform-domain').parent('.form-group').removeClass('has-error');
                     $('#autoform-form').hide();
                     $('#autoform-form-sent').show();
-
-                } else {
+                })
+                .fail(function() {
                     $('#autoform-form-error').show();
-                }
-            });
+                });
 
             plausible('Registration', {props: {product: 'Autoform'}});
             return true;
         }
     });
-});
\ No newline at end of file
+
+    const wrapper = document.getElementById('flash-message-wrapper')
+    wrapper.addEventListener('click', (event) => {
+        const isButton = event.target.nodeName === 'SPAN';
+        if (!isButton) {return;}
+
+        $('#autoform-email').parent('.form-group').removeClass('has-error');
+        $('#autoform-domain').parent('.form-group').removeClass('has-error');
+
+        if(event.target.className === 'error') {
+            $('#autoform-error').hide();
+        }
+        else if(event.target.className === 'form-error') {
+            $('#autoform-form-error').hide();
+        }
+        else if(event.target.className === 'success') {
+            $('#autoform-form-sent').hide();
+        }
```

Comment:
Podľa mňa pri success to netreba meniť, lebo tam nie je potrebné, aby user vedel zavrieť a potom znova vidieť oznam - registruje sa iba raz a potom ide preč. A keď už, tak vieš vytvoriť v JS funkciu napr. addCloseButtonListener(node), kam pošleš jeden flash message (autoform-error, autoform-form-error, autoform-form-sent), a v tej funkcii nastavíš listener na jeho vnorený button X. Potom nemusíš robiť takéto všelijaké ify.